adyenofficial.php

```diff
@@ -2194,4 +2213,22 @@ private function getService($serviceName)
     {
         return \Adyen\PrestaShop\service\adapter\classes\ServiceLocator::get($serviceName);
     }
+
+    /**
+     * Check if payment methods have already been obtained in this object. If not, fetch them.
+     *
+     * @return array
+     * @throws Exception
+     */
+    private function getPaymentMethods()
+    {
+        if (empty($this->paymentMethods)) {
```

Comment:
In case the details changed like country code / amount / currency but the paymentMethods call has been done already, are we going to perform another payment methods again?

Reply:
Yes, in that case the payment methods will be fetched again, since the class is instantiated on every step. So if the user is in the payment methods page, and goes back to change something, class will be re-instantiated which would trigger the fetch call from Adyen